modules/memfault-firmware-sdk/memfault_gnss_metrics.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <modem/location.h>
+#include "memfault/metrics/metrics.h"
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(memfault_ncs_metrics, CONFIG_MEMFAULT_NCS_LOG_LEVEL);
+
+static bool session_in_progress = false;
+static location_event_handler_t real_location_event_handler;
+
+static void memfault_location_event_handler(const struct location_event_data *event_data);
+static void memfault_gnss_metrics_start_fix_session(void);
+static void memfault_gnss_metrics_stop_fix_session(const struct location_event_data *event_data);
+
+// To maintain separation of metrics collection from NCS library code, we need a way to
+// know when a fix request is made, and when the result of the request is ready (fix or error).
+// There is a location event handler that indicates a location request result, but only one,
+// which should be reserved for application code. There is also no callback mechanism to
```

Comment:
We could add a list of event handlers into Location library similarly to lte_lc.
I can handle this if that's decided to be the way to go.

Reply:
Yes I believe this would be the cleanest approach. That would be great if you'd be open to adding this functionality!